### PR TITLE
LPS-138924 Semver

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-api/src/main/java/com/liferay/portal/workflow/metrics/search/index/ProcessWorkflowMetricsIndexer.java [hunk 1]
```diff
@@ -25,6 +26,11 @@
  */
 public interface ProcessWorkflowMetricsIndexer {
 
+	public Document addProcess(AddProcessRequest addProcessRequest);
+
+	public Document addProcess(
+		AddProcessRequest addProcessRequest, String[] versions);
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-api/src/main/java/com/liferay/portal/workflow/metrics/search/index/ProcessWorkflowMetricsIndexer.java:30]: Hi @Selton, join these methods

thanks!

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java [hunk 2]
```diff
@@ -124,6 +128,38 @@ public AddNodeRequest createAddNodeRequest(
...
+		).description(
+			kaleoDefinition.getDescription()
+		).modifiedDate(
+			kaleoDefinition.getModifiedDate()
+		).name(
+			kaleoDefinition.getName()
+		).processId(
+			kaleoDefinition.getKaleoDefinitionId()
+		).title(
+			kaleoDefinition.getTitle(defaultLanguageId)
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java:156]: @seltong, set the content of defaultLanguageId variable inline, please.

Thanks!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
Just started reviewing :)

**Comment by seltong** [comment]
Just started reviewing :)

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Comment by seltong** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/helper/WorkflowMetricsRESTTestHelper.java [hunk 3]
```diff
@@ -1009,15 +1028,36 @@ public void deleteTasks(long companyId, long processId) throws Exception {
 	}
 
 	public void restoreProcess(Document document) throws Exception {
+		AddProcessRequest.Builder builder = new AddProcessRequest.Builder();
+		String[] versions = {
+			StringBundler.concat(
+				document.getString("version"), CharPool.PERIOD, 0)
+		};
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/helper/WorkflowMetricsRESTTestHelper.java:1045]: Hi @seltong,

What do you think this approach below:

`AddProcessRequest.Builder builder = new AddProcessRequest.Builder();`

`builder.active ... titleMap( ...`

`String version = StringBundler.concat(document.getString("version"), CharPool.PERIOD, 0);`

`builder.version(version).versions(new String[] { version })`

`_processWorkflowMetricsIndexer.addProcess(builder.build());`

slack me monday

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/helper/WorkflowMetricsRESTTestHelper.java (hunk 3 above)
Comment (MarinhoFeliphe) [reply modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/helper/WorkflowMetricsRESTTestHelper.java:1045]: Just to bring the version variable more closer to its use

**Comment by MarinhoFeliphe** [comment]
BTW, very good pull! :D

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/helper/WorkflowMetricsRESTTestHelper.java (hunk 3 above)
Comment (seltong) [reply modules/dxp/apps/portal-workflow/portal-workflow-metrics-rest-test/src/testIntegration/java/com/liferay/portal/workflow/metrics/rest/resource/v1_0/test/helper/WorkflowMetricsRESTTestHelper.java:1045]: Likes good for me. I talk to you about it on Monday

**Comment by seltong** [comment]
ci:test:sf

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java [hunk 4]
```diff
@@ -124,6 +129,65 @@ public AddNodeRequest createAddNodeRequest(
...
+					).map(
+						KaleoDefinitionVersion::getVersion
+					).toArray(
+						String[]::new
+					)
+				).build();
+			}
+		}
+		catch (PortalException portalException) {
+			portalException.printStackTrace();
```
Comment (MarinhoFeliphe) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java:185]: Hi @seltong, use `_log.warn...` instead `portalException.printStackTrace();` and before to log a warn check if warn is enabled `_log.isWarnEnabled()`

Thank you!

**Pull request closed without merging**